src/metabase/api/field.clj

```diff
@@ -246,47 +247,35 @@
   "Fetch FieldValues, if they exist, for a `field` and return them in an appropriate format for public/embedded
   use-cases."
   [{has-field-values-type :has_field_values, field-id :id, has_more_values :has_more_values, :as field}]
-  ;; if there's a human-readable remapping, we need to do all sorts of nonsense to make this work and return pairs of
-  ;; `[original remapped]`. The code for this exists in the [[search-values]] function below. So let's just use
-  ;; [[search-values]] without a search term to fetch all values.
-  (if-let [human-readable-field-id (when (= has-field-values-type :list)
-                                     (t2/select-one-fn :human_readable_field_id Dimension :field_id (u/the-id field)))]
+  ;; TODO: explain why using remapped fields is restricted to `has_field_values=list`
+  (if-let [remapped-field-id (when (= has-field-values-type :list)
+                               (chain-filter/remapped-field-id field-id))]
     {:values          (search-values (api/check-404 field)
-                                     (api/check-404 (t2/select-one Field :id human-readable-field-id)))
+                                     (api/check-404 (t2/select-one Field :id remapped-field-id)))
      :field_id        field-id
      :has_more_values has_more_values}
     (params.field-values/get-or-create-field-values-for-current-user! (api/check-404 field))))
 
-(defn check-perms-and-return-field-values
-  "Impl for `GET /api/field/:id/values` endpoint; check whether current user has read perms for Field with `id`, and, if
-  so, return its values."
-  [field-id]
-  (let [field (api/check-404 (t2/select-one Field :id field-id))]
-    (api/check-403 (params.field-values/current-user-can-fetch-field-values? field))
-    (field->values field)))
-
-;; todo: we need to unify and untangle this stuff
 (defn field-id->values
```

Comment:
this is a bit confuse given we have `field->values` too.  and `field-id->values` is only used n `api.card`, maybe we should move its definition there?

Reply:
Renamed this function to `search-values-from-field-id`. I don't really feel like moving the function to `api.card` is an improvement because it feels more closely coupled with other functions from this namespace like `search-values` and `field->values`. Eventually all the `FieldValues` should probably move to a common namespace so we don't feel like splitting logic between api namespaces